resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/domain/CompletenessEntity.java

```diff
@@ -1,111 +1,111 @@
-            this.entityType = entityType;
-            this.entityId = entityId;
-        }
-
-        @Override
-        @Generated
-        public boolean equals(final Object o) {
-            if (this == o) {
-                return true;
-            }
-            if (!(o instanceof CompletenessId)) {
-                return false;
-            }
-            final CompletenessId that = (CompletenessId) o;
-            return entityType == that.entityType &&
-                    Objects.equals(entityId, that.entityId);
-        }
-
-        @Override
-        @Generated
-        public int hashCode() {
-            return Objects.hash(entityType, entityId);
-        }
-    }
-}
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.resource.domain;
+
+import org.springframework.data.annotation.PersistenceConstructor;
+import org.springframework.data.redis.core.RedisHash;
+
+import uk.gov.gchq.palisade.Generated;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.Table;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+@Entity
+@Table(name = "completeness",
+        indexes = {
+                @Index(name = "id_pair_hash", columnList = "id_pair_hash", unique = true),
+                @Index(name = "entity_type", columnList = "entity_type"),
+                @Index(name = "entity_id", columnList = "entity_id"),
+        })
+@RedisHash("CompletenessEntity")
+public class CompletenessEntity implements Serializable {
+    @Id
+    @org.springframework.data.annotation.Id
+    @Column(name = "id_pair_hash", columnDefinition = "integer", nullable = false)
+    private Integer id;
+
+    @Column(name = "entity_type", columnDefinition = "integer", nullable = false)
+    @Enumerated(EnumType.ORDINAL)
+    protected EntityType entityType;
+
+    @Column(name = "entity_id", columnDefinition = "varchar(255)", nullable = false)
+    protected String entityId;
+
+    public CompletenessEntity() {
+    }
+
+    @PersistenceConstructor
+    public CompletenessEntity(final EntityType entityType, final String entityId) {
+        this.entityType = entityType;
+        this.entityId = entityId;
+        this.id = hashCode();
```

Comment:
Could see this being entityType.toString() + "::" + entityId

Reply:
I'm going to change this as SQ doesn't like using `hash()` in the constructor.

Reply:
The implemented change caused a test issue. Left it for now.